flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java

```diff
@@ -898,6 +825,116 @@ public void testSlotSharingOnAllVerticesInSameSlotSharingGroupByDefaultDisabled(
+		streamGraph.setGlobalDataExchangeMode(GlobalDataExchangeMode.FORWARD_EDGES_PIPELINED);
+		final JobGraph jobGraph = StreamingJobGraphGenerator.createJobGraph(streamGraph);
+
+		final List<JobVertex> verticesSorted = jobGraph.getVerticesSortedTopologicallyFromSources();
+		final JobVertex sourceVertex = verticesSorted.get(0);
+		final JobVertex map1Vertex = verticesSorted.get(1);
+		final JobVertex map2Vertex = verticesSorted.get(2);
+
+		assertEquals(ResultPartitionType.PIPELINED_BOUNDED, sourceVertex.getProducedDataSets().get(0).getResultType());
+		assertEquals(ResultPartitionType.BLOCKING, map1Vertex.getProducedDataSets().get(0).getResultType());
+		assertEquals(ResultPartitionType.BLOCKING, map2Vertex.getProducedDataSets().get(0).getResultType());
+	}
+
+	@Test
+	public void testPointwiseEdgesPipelinedMode() {
+		final StreamGraph streamGraph = createStreamGraphForGlobalDataExchangeModeTests();
+		streamGraph.setGlobalDataExchangeMode(GlobalDataExchangeMode.POINTWISE_EDGES_PIPELINED);
+		final JobGraph jobGraph = StreamingJobGraphGenerator.createJobGraph(streamGraph);
+
+		final List<JobVertex> verticesSorted = jobGraph.getVerticesSortedTopologicallyFromSources();
+		final JobVertex sourceVertex = verticesSorted.get(0);
+		final JobVertex map1Vertex = verticesSorted.get(1);
+		final JobVertex map2Vertex = verticesSorted.get(2);
+
+		assertEquals(ResultPartitionType.PIPELINED_BOUNDED, sourceVertex.getProducedDataSets().get(0).getResultType());
+		assertEquals(ResultPartitionType.PIPELINED_BOUNDED, map1Vertex.getProducedDataSets().get(0).getResultType());
+		assertEquals(ResultPartitionType.BLOCKING, map2Vertex.getProducedDataSets().get(0).getResultType());
+	}
+
+	@Test
+	public void testGlobalDataExchangeModeDoesNotOverrideSpecifiedShuffleMode() {
+		final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		final DataStream<Integer> source = env.fromElements(1, 2, 3).setParallelism(1);
+		final DataStream<Integer> forward = new DataStream<>(env, new PartitionTransformation<>(
+			source.getTransformation(), new ForwardPartitioner<>(), ShuffleMode.PIPELINED));
+		forward.map(i -> i).startNewChain().setParallelism(1);
+		final StreamGraph streamGraph = env.getStreamGraph();
+		streamGraph.setGlobalDataExchangeMode(GlobalDataExchangeMode.ALL_EDGES_BLOCKING);
+
+		final JobGraph jobGraph = StreamingJobGraphGenerator.createJobGraph(streamGraph);
+
+		final List<JobVertex> verticesSorted = jobGraph.getVerticesSortedTopologicallyFromSources();
+		final JobVertex sourceVertex = verticesSorted.get(0);
+
+		assertEquals(ResultPartitionType.PIPELINED_BOUNDED, sourceVertex.getProducedDataSets().get(0).getResultType());
+	}
+
+	/**
+	 * Topology: source(parallelism=1) --(forward)--> map1(parallelism=1)
+	 *           --(rescale)--> map2(parallelism=2) --(rebalance)--> sink(parallelism=2).
+	 */
+	private StreamGraph createStreamGraphForGlobalDataExchangeModeTests() {
```

Comment:
Maybe it makes sense to move the added tests to a new class.
Also, can be declared `static`.

Reply:
Ok. I have moved the tests to `StreamingJobGraphGeneratorWithGlobalDataExchangeModeTest`.
Also make this method static.